### PR TITLE
Feature/restore nugraph2

### PR DESCRIPTION
**Pull request by cerati**
This PR fixes and improves NuGraph2 in the main branch. 
It contains the following changes:

1. Fix usage of InputNorm for NuGraph2 (main problem) and other small changes to avoid syntax errors or crashed
2. Change scripts/train.py to be able to train NuGraph2 with a command line option instead of changing the file
3. Fix feature extension, which was crashing in NuGraph2 and silently failing in NuGraph3
4. Significantly speedup NuGraph2 training: restructure linear.py to use a unique tensor for all instances of the latent space (one per semantic class) instead of N different tensors, allowing for simultaneous operations instead of loop over categories; use of torch compile for plane_net and nexus_net; compute shortcut tensors only once (minor impact); consistently use 8 workers as default (also impacting NuGraph3)

### CODE REVIEW

File: nugraph/nugraph/models/nugraph2/decoders.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def forward(self, x: TD, batch: TD) -> dict[str, TD]: # pylint: disable=unused-a
     def arrange(self, data: NuGraphData) -> tuple[T, T]:
         x = torch.cat([data[p].x_semantic for p in self.planes], dim=0)
         y = torch.cat([data[p].y_semantic for p in self.planes], dim=0)
+        y[y >= len(self.classes)] = -1
```
Comment (vhewes) [inline nugraph/nugraph/models/nugraph2/decoders.py:147]: what's the purpose of this modification? do we expect the semantic truth to contain label integers greater than the number of classes?

Comment (cerati) [reply nugraph/nugraph/models/nugraph2/decoders.py:147]: in general no, but I do have some old training files I still use that have such a problem. I'd be happy to include this protection in the code, but if you prefer not to I have other options to deal with the problem

Comment (vhewes) [reply nugraph/nugraph/models/nugraph2/decoders.py:147]: okay! i'm concerned about cases where there are other pathologies in training files that won't be caught because of this line, but i don't really use nugraph2 any more so i don't feel strongly. i'll approve the PR so you can merge as-is if you like. thank you!!

**Review by vhewes — APPROVED** [review]
